src/peer_manager.rs

```diff
@@ -314,9 +317,9 @@ impl Candidate {
             | Candidate::ResourceProof {
                 res_proof_start, ..
             } => {
-                // TODO: need better fix. Using a larger timeout to allow Online to accumulate via gossip
-                // than the prev timeout for grp-msg accumulation.
-                res_proof_start.elapsed() > RESOURCE_PROOF_DURATION + ACCUMULATION_TIMEOUT * 3
+                // TODO: need better fix. Using a larger timeout to allow Online to accumulate via
+                // gossip than the prev timeout for grp-msg accumulation.
+                res_proof_start.elapsed() > RESOURCE_PROOF_DURATION + CANDIDATE_EXPIRED_TIMEOUT
```

Comment:
I am not 100% sure about the name or the comment explaining the name.
With the name and comment, it feels like it should read `res_proof_start.elapsed() > CANDIDATE_EXPIRED_TIMEOUT`.

Maybe you could have CANDIDATE_EXPIRED_TIMEOUT = RESOURCE_PROOF_DURATION + Duration::from_secs(90);